### PR TITLE
Fix macaroon dist typying

### LINKED ISSUE #55: Compiled typescript is failing
When building a project like `jujulib` we are getting the following output:

```console
$ yarn build
yarn run v1.22.19
$ tsc
node_modules/@canonical/macaroon-bakery/dist/bakery.d.ts:12:66 - error TS7016: Could not find a declaration file for module 'macaroon'. '/home/goulin/Lab/juju/js-libjuju/node_modules/macaroon/macaroon.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/macaroon` if it exists or add a new declaration (.d.ts) file containing `declare module 'macaroon';`

12 import { Macaroon, Error as MacaroonError, MacaroonObject } from "macaroon";
                                                                    ~~~~~~~~~~


Found 1 error in node_modules/@canonical/macaroon-bakery/dist/bakery.d.ts:12

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### PR DESCRIPTION
**Pull request by goulinkh**
## Description

- Fix the Typescript error
- Copy the eslint and prettier config from the jujulib project
- Add Github CI

## Issues
 Fixes #55

### CODE REVIEW

**Comment by huwshimi** [comment]
@goulinkh I linked bakeryjs in jujulib and when I run `yarn build` in jujulib linked to bakeryjs master I see the same error as the original issue, but with this branch I get an error:

```
api/client.ts:308:53 - error TS2345: Argument of type '(macaroons: Macaroon[]) => void' is not assignable to parameter of type '(macaroon: MacaroonObject[]) => void'.
  Types of parameters 'macaroons' and 'macaroon' are incompatible.
    Type 'MacaroonObject[]' is not assignable to type 'Macaroon[]'.
      Type 'MacaroonObject' is not assignable to type 'Macaroon'.
        Type 'string' is not assignable to type 'Macaroon'.

308           this._bakery.discharge(dischargeRequired, onSuccess, onFailure);
                                                        ~~~~~~~~~


Found 1 error in api/client.ts:308
```

**Comment by goulinkh** [comment]
> @goulinkh I linked bakeryjs in jujulib and when I run `yarn build` in jujulib linked to bakeryjs master I see the same error as the original issue, but with this branch I get an error:
> 
> ```
> api/client.ts:308:53 - error TS2345: Argument of type '(macaroons: Macaroon[]) => void' is not assignable to parameter of type '(macaroon: MacaroonObject[]) => void'.
>   Types of parameters 'macaroons' and 'macaroon' are incompatible.
>     Type 'MacaroonObject[]' is not assignable to type 'Macaroon[]'.
>       Type 'MacaroonObject' is not assignable to type 'Macaroon'.
>         Type 'string' is not assignable to type 'Macaroon'.
> 
> 308           this._bakery.discharge(dischargeRequired, onSuccess, onFailure);
>                                                         ~~~~~~~~~
> 
> 
> Found 1 error in api/client.ts:308
> ```

Thanks for the QA @huwshimi ! I've filed a [PR](https://github.com/juju/js-libjuju/pull/103) to addess this issue.

File: src/bakery.ts [hunk 1]
```diff
@@ -12,17 +12,17 @@
  * ```
  */
 
-import macaroonlib, {
-  Macaroon,
-  Error as MacaroonError,
-  MacaroonObject,
-} from "macaroon";
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
```
Comment (huwshimi) [inline src/bakery.ts]: If I remove this ts-ignore I don't seem to get any errors. What was the issue here?

**Review by huwshimi — APPROVED** [review]

File: src/bakery.ts (hunk 1 above)
Comment (goulinkh) [reply src/bakery.ts]: Nice catch! this is a trace of my fight with TS :)

**Pull request merged**